### ISSUE TITLE
val: Scanning (dataset path).cache ... taking so long during training

### ISSUE THREAD

**Issue by guillermovc** [ultralytics/ultralytics#3794]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Hi, im trying to combine optuna with W&B to make different trials changing parameters, but the scanning of my dataset is taking so long (even stucked). The weird thing is that im using a very small dataset (351 images total). My dataset is loaded locally.

I hope you can help me out. Thanks!

![image](https://github.com/ultralytics/ultralytics/assets/43889001/fca60045-320a-48e3-b260-0db60410f13c)

### Environment

- Using conda
- YOLO ultralytics 8.0.123
- Optuna 3.2.0
- Weights & Biases 0.15.4
- Python 3.10.11
- Windows 11
- GPU: RTX 2060 12GB
- RAM: 16GB
- M2 SSD
- CPU Ryzen 5 6-cores

### Minimal Reproducible Example
```
from ultralytics import YOLO

from pathlib import Path
import optuna
import os
os.environ["KMP_DUPLICATE_LIB_OK"]="TRUE"
import pickle

from optuna.integration.wandb import WeightsAndBiasesCallback

def objective(trial):

    custom_cfg = {
        "model": [
            "yolov8m.pt", 
            # "yolov8l.pt",
        ],
        "imgsz": [640],
        "batch": [8, 16],
        
        "optimizer": [
            "SGD", 
            # "Adam", 
            "Adamax", 
            "AdamW", 
        #   "NAdam", 
            "RAdam", 
            # "RMSProp",
        ],
        
        # Augmentations
        "hsv_s": [0.7, 0.35, 0.0],
        "hsv_v": [0.4, 0.2, 0.0],
        "scale": [0.5, 0.3, 0.0],
        "fliplr": [0.5, 0.0],
    }
    
    # 2. Suggest values of the hyperparameters using a trial object.
    model_name = trial.suggest_categorical("model", custom_cfg["model"])
    imgsz = trial.suggest_categorical("imgsz", custom_cfg["imgsz"])
    batch = trial.suggest_categorical("batch", custom_cfg["batch"])
    optimizer = trial.suggest_categorical("optimizer", custom_cfg["optimizer"])
    hsv_s = trial.suggest_categorical("hsv_s", custom_cfg["hsv_s"])
    hsv_v = trial.suggest_categorical("hsv_v", custom_cfg["hsv_v"])
    scale = trial.suggest_categorical("scale", custom_cfg["scale"])
    fliplr = trial.suggest_categorical("fliplr", custom_cfg["fliplr"])

    # Train
    model = YOLO(model_name)

    DATA = r"C:\Macrodata\Macrodata_libretas\data\training_17jul.yaml"
    EPOCHS = 5
    EXTRA = "17jul"
    PROJECT = "hptunning_17jul"
    NAME = f"{EXTRA}_{model_name}_imgsz{imgsz}_bs{batch}_{EPOCHS}ep_opt{optimizer}_hsv_s{hsv_s}_hsv_v{hsv_v}_scale{scale}_fliplr{fliplr}"

    model.train(
        data = DATA, 
        epochs = EPOCHS, 
        name = NAME,
        exist_ok=True,
        project=PROJECT,

        imgsz=imgsz,
        batch=batch,
        optimizer=optimizer,
        hsv_s=hsv_s,
        hsv_v=hsv_v,
        scale=scale,
        fliplr=fliplr,
    )

    best_path = Path(PROJECT) / Path(NAME) / "weights" / "best.pt"
    best_model = YOLO(best_path)
    # Validate
    r = best_model.val(conf=0.65, iou=0.6, device=0,)

    maP50_95 = r.results_dict["metrics/mAP50-95(B)"]

    return maP50_95

if __name__ == '__main__':

    from wandb_conf import wandb_project_name, wand_token

    os.environ["WANDB_API_KEY"] = wand_token

    wandb_kwargs = {
        "project": wandb_project_name,
        "api_key": wand_token,
    }

    wandbc = WeightsAndBiasesCallback(wandb_kwargs=wandb_kwargs, as_multirun=True)
    study = optuna.create_study(direction='maximize')
    study.optimize(objective, n_trials=30, callbacks=[wandbc])
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Kayzwer** [comment]
@guillermovc try to delete the .cache in you dataset and run it again, maybe the cache file corrupted.

**Comment by glenn-jocher** [comment]
Hi @guillermovc,

Thanks for reaching out with your issue. The problem you're experiencing might be due to a corrupted .cache file from your dataset. Here's what you can do: try deleting the .cache files that are present in your dataset directory. Once you've done this, rerun your program. This typically fixes any issues related to the lengthy scanning of .cache files.

Please let us know whether this resolves your issue. We're here to help, so if you need further assistance, don't hesitate to ask.

Best Regards,
Glenn Jocher and the Ultralytics team.

**Comment by guillermovc** [comment]
Hi guys! @glenn-jocher , @Kayzwer , thanks for your answers.

I will delete the .cache file and try it out again. 
Since im runing several experiments with `optuna`, do you recommend me to implment a logic to delete the .cache file when starting each of these? is this .cache file using during training or just used when you start a new training?

**Comment by glenn-jocher** [comment]
@Kayzwer hello,

The .cache file is generated for speed improvement, especially in the case of large datasets. It keeps track of your images and labels, so the program doesn't have to rescan every time it needs to access this data. This speeds up things considerably when dealing with larger datasets.

If you make modifications to your images or labels, or if a .cache file gets corrupted, you might need to delete it so that a new one can be generated when you rerun your program. But if your training data remains unchanged and no corruption has occurred, there's no need to delete the .cache file for every new experiment - it's meant to be reused to speed things up.

Running several trials with `Optuna` should require changes to the hyperparameters, not the dataset, therefore, it should not impact the .cache file. So you shouldn't need to delete it for every trial. The file is generated at the start of the training and used throughout the entire process.

However, in practical terms, if you're dealing with a smaller dataset and finding `Optuna` is stuck due to caching issues, you could try to disable caching.

I trust this answers your query. Let us know if you need further assistance!

Best Regards,
Glenn Jocher and the Ultralytics team.
